src/utils/mod.rs

```diff
@@ -886,3 +913,66 @@ impl TryFrom<TpmtTkVerified> for TPMT_TK_VERIFIED {
         })
     }
 }
+
+/// Create the TPM2B_PUBLIC structure for a restricted decryption key.
```

Comment:
By "decryption key", do you mean a public key to verify signatures?

Reply:
Or does it do both?

Reply:
From the spec:
>A hierarchy is rooted in a secret seed key, kept in the TPM. To create a hierarchy of keys, the seed key (Primary Seed) is used to generate a key that uses a specific set of algorithms. If this key is a **restricted decryption key**, then it is a Parent  Key. If it  is not a Derivation Parent, then  it is  a Storage Parent under which other objects may be created or attached.A Storage Parent provides protection for the sensitive area in another object (a child) when that object is stored  outside  of  the  TPM.  Protection  is  provided  by  symmetric  encryption  and  HMAC-based  integrity protection of the sensitive area. There are two different cases for sensitivearea encryption: storage and duplication.